app/models/accounting/quickbooks/transaction_extractor.rb

```diff
@@ -34,11 +34,13 @@ def extract_line_items
           acct = Accounting::Account.find_by(qb_id: li[qb_li_detail_key]['account_ref']['value'])
           # skip if line item does not have an account in Madeline
           next unless acct
+          posting_type = li[qb_li_detail_key]['posting_type']
+          posting_type ||= existing_li_posting_type unless posting_type.present?
```

Comment:
Curious for my ruby learning - when does "unless present" have an effect here? I understood ||= to be the same.

Reply:
also not super impt to answer this right now.

Reply:
@DevneyHamilton when `posting_type` is an empty string or an empty array or hash rather than a nil. It should probably just be a normal `=` with the unless, I can fix that.

Reply:
ah, thanks!